tests/Aspire.Hosting.Tests/Azure/AzureBicepResourceTests.cs

```diff
@@ -116,6 +116,24 @@ public async Task AssertManifestLayout()
         Assert.Equal(expectedManifest, manifest.ToString());
     }
 
+    [Fact]
+    public async Task AddAzureCosmosDBEmulator()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+
+        var cosmos = builder.AddAzureCosmosDB("cosmos").RunAsEmulator(e =>
+        {
+            e.WithEndpoint("emulator", e => e.AllocatedEndpoint = new(e, "localost", 10001));
+        });
+
+        Assert.True(cosmos.Resource.IsContainer());
+
+        var cs = AzureCosmosDBEmulatorConnectionString.Create(10001);
+
+        Assert.Equal(cs, cosmos.Resource.ConnectionStringExpression.ValueExpression);
+        Assert.Equal(cs, await ((IResourceWithConnectionString)cosmos.Resource).GetConnectionStringAsync());
```

Comment:
It's odd that these 2 strings are the same. I would have thought the ValueExpression wouldn't have the port in it.

Reply:
Yes this is strange. It's because we don't build the right connection string for the manifest for these emulators. I'll look at it in a follow up PR (just for clarity)